pkg/filepos/position.go

```diff
@@ -122,3 +123,32 @@ func (p *Position) IsNextTo(otherPostion *Position) bool {
 	}
 	return false
 }
+
+// PopulateAnnotationPositionFromNode takes an annotation's position, the position of the node it annotates,
+// and the comments from the same node, and creates an approximation of the annotation file and line string.
+func PopulateAnnotationPositionFromNode(annPos *Position, nodePosition *Position, nodeComments []Meta) *Position {
+	leftPadding := 0
+	if nodePosition.IsKnown() {
+		nodeLine := nodePosition.GetLine()
+		leftPadding = len(nodeLine) - len(strings.TrimLeft(nodeLine, " "))
+	}
+
+	lineString := ""
+	for _, c := range nodeComments {
+		if c.Position.IsKnown() && c.Position.AsIntString() == fmt.Sprintf("%d", annPos.LineNum()) {
+			lineString = fmt.Sprintf("%v#%s", strings.Repeat(" ", leftPadding), c.Data)
+		}
+	}
+
+	annPos.SetFile(nodePosition.GetFile())
+	annPos.SetLine(lineString)
+
+	return annPos
```

Comment:
[Thinking out loud here] I wonder if these changes to add annotation positions would benefit from or are risky enough for a test. I know we have some file position tests in `template_test.go`.